tls/bignum.c

```diff
@@ -1446,3 +1442,15 @@ ttls_mpi_inv_mod(TlsMpi *X, const TlsMpi *A, const TlsMpi *N)
 
 	ttls_mpi_pool_cleanup_ctx((unsigned long)TA, false);
 }
+
+/**
+ * Calculate X = X^2 mod N.
+ */
+void
+ttls_mpi_sq_mod(TlsMpi *X, const TlsMpi *N)
```

Comment:
The function name and comment are misleading. According to Handbook of Applied Cryptography 14.85 algorithm (I strongly recommend to download the book along with BigNun Math by Denis - these books are very useful for the TLS crypt development) this is ` A = 1; A = A ^ 2`. However, I can't explain why do we make the left shift - probably because we slide in opposite direction....

Anyway this is not a generic big integer squaring, but a pure initialization of a helping constant for the exponentiation computation.

Also it'd be good to place the helping computation function somewhere on the top of the file.

Reply:
Fixed